src/EditorFeatures/CSharpTest/RemoveUnnecessaryParentheses/RemoveUnnecessaryParenthesesTests.cs

```diff
@@ -2280,5 +2287,86 @@ void Test(bool a)
     }
 }", new TestParameters(options: RemoveAllUnnecessaryParentheses));
         }
+
+        [WorkItem(27925, "https://github.com/dotnet/roslyn/issues/27925")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestUnnecessaryParenthesisDiagnosticSingleLineExpression()
+        {
+            var openParenthesesDiagnostic = GetRemoveUnnecessaryParenthesesDiagnostic("(", 4, 16, true);
```

Comment:
Still investigating the best way to address this one.  I'm currently looking for a good way to handle nested markup spans (since there can be multiple diagnostics on the same character(s) but with different spans).

It seems possible with a little work to add/retrieve multiple spans from the document, but then the question is how to associate a particular span with the specific diagnostic that should be in the span.  If you have any ideas I'd be happy to hear them.